app/components/root/root.js

```diff
@@ -29,21 +40,98 @@ export default class Root extends React.Component {
     componentDidMount() {
         AppState.addEventListener('change', this.handleAppStateChange);
         EventEmitter.on(Constants.CONFIG_CHANGED, this.handleConfigChanged);
+        MessageBarManager.registerMessageBar(this.refs.notification);
+        this.configurePushNotifications();
     }
 
     componentWillUnmount() {
         AppState.removeEventListener('change', this.handleAppStateChange);
         EventEmitter.off(Constants.CONFIG_CHANGED, this.handleConfigChanged);
+        MessageBarManager.unregisterMessageBar();
+        PushNotification.unregister();
     }
 
+    onRegisterDevice = (data) => {
+        const prefix = Platform.OS === 'ios' ? Constants.PUSH_NOTIFY_APPLE_REACT_NATIVE : Constants.PUSH_NOTIFY_ANDROID_REACT_NATIVE;
+        this.props.actions.setDeviceToken(`${prefix}:${data.token}`);
+    };
+
+    onPushNotification = (notification) => {
+        const {foreground, userInteraction, data, message} = notification;
+        let noty;
```

Comment:
Is `noty` supposed to be short for `notification`? Maybe you could rename the parameter to `deviceNotification` and `noty` to `notification` so that its name is the same in `onPushNotification` and the two methods it calls